x-pack/plugin/ilm/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java

```diff
@@ -0,0 +1,135 @@
+        return Settings.builder()
+            .put(ThreadContext.PREFIX + ".Authorization", token)
+            .build();
+    }
+
+    @Before
+    public void init() throws Exception {
+        Request request = new Request("PUT", "/_cluster/settings");
+        XContentBuilder pollIntervalEntity = JsonXContent.contentBuilder();
+        pollIntervalEntity.startObject();
+        {
+            pollIntervalEntity.startObject("transient");
+            {
+                pollIntervalEntity.field(LifecycleSettings.LIFECYCLE_POLL_INTERVAL, "1s");
+            }pollIntervalEntity.endObject();
+        } pollIntervalEntity.endObject();
+        request.setJsonEntity(Strings.toString(pollIntervalEntity));
+        assertOK(adminClient().performRequest(request));
+        indexSettingsWithPolicy = Settings.builder()
+            .put(LifecycleSettings.LIFECYCLE_NAME, deletePolicy)
+            .put("number_of_shards", 1)
+            .put("number_of_replicas", 0)
+            .build();
+        createNewSingletonPolicy(deletePolicy,"delete", new DeleteAction());
+    }
+
+    public void testCanManageIndexAndPolicyDifferentUsers() throws Exception {
+        String index = "ilm-00001";
+        createIndexAsAdmin(index, indexSettingsWithPolicy, "");
+        assertBusy(() -> assertFalse(indexExists(index)));
+    }
+
+    /**
+     * This tests the awkward behavior where an admin can have permissions to create a policy,
+     * but then not have permissions to operate on an index that was later associated with that policy by another
+     * user
+     */
+    @SuppressWarnings("unchecked")
+    public void testCanManageIndexWithNoPermissions() throws Exception {
+        createIndexAsAdmin("not-ilm", indexSettingsWithPolicy, "");
+        Request request = new Request("GET", "/not-ilm/_ilm/explain");
+        // test_ilm user does not have permissions on this index
+        ResponseException exception = expectThrows(ResponseException.class, () -> client().performRequest(request));
+        assertThat(exception.getResponse().getStatusLine().getStatusCode(), equalTo(RestStatus.FORBIDDEN.getStatus()));
+
+        assertBusy(() -> {
+            Response response = adminClient().performRequest(request);
+            assertOK(response);
+            try (InputStream is = response.getEntity().getContent()) {
+                Map<String, Object> mapResponse = XContentHelper.convertToMap(XContentType.JSON.xContent(), is, true);
+                Map<String, Object> indexExplain = (Map<String, Object>) ((Map<String, Object>) mapResponse.get("indices")).get("not-ilm");
+                assertThat(indexExplain.get("managed"), equalTo(true));
+                assertThat(indexExplain.get("step"), equalTo("error"));
+                assertThat(indexExplain.get("failed_step"), equalTo("readonly"));
+                assertThat(indexExplain.get("step_info"), equalTo("permissionsss!"));
```

Comment:
wat?

Do we really assert that the explanation for the step is "permissionsss!" :snake: ?

Reply:
haha, yes!

This is technically WIP since the test does not pass... but I have no idea why. I was hoping to at least get early feedback on the types of tests... as to why things are stuck on `readonly`. still a mystery to me and I am debugging it

Reply:
I can help look into this tomorrow if you are still having trouble finding the cause of the failure